src/PaymentGateways/Gateways/Stripe/Actions/UpdateStripeFormBuilderSettingsMeta.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\Stripe\Actions;
+
+use Give\DonationForms\Models\DonationForm;
+
+/**
+ * Update Stripe settings on backwards compatible form meta.
+ *
+ * @since 3.0.0
+ */
+class UpdateStripeFormBuilderSettingsMeta
+{
+    /**
+     * @since 3.0.0
+     * @param  DonationForm  $form
+     */
+    public function __invoke(DonationForm $form)
+    {
+        $paymentGatewaysBlock = $form->blocks->findByName('givewp/payment-gateways');
+        if (!$paymentGatewaysBlock) {
+            give()->form_meta->delete_meta($form->id, 'give_stripe_per_form_accounts');
+            give()->form_meta->delete_meta($form->id, '_give_stripe_default_account');
+
+            return;
+        }
+
+        [
+            'stripeUseGlobalDefault' => $stripeUseGlobalDefault,
+            'stripeAccountId' => $stripeAccountId
+        ] = $paymentGatewaysBlock->getAttributes() + ['stripeUseGlobalDefault' => true, 'stripeAccountId' => ''];
+
+        if (is_bool($stripeUseGlobalDefault)) {
+            give()->form_meta->update_meta($form->id, "give_stripe_per_form_accounts", $stripeUseGlobalDefault ? 'disabled' : 'enabled');
+        }
```

Comment:
How about instead of `is_bool`, you do:
```
filter_var($stripeUseGlobalDefault, FILTER_VALIDATE_BOOLEAN)
```
to get the boolean value.

Reply:
Here I'm just guaranteeing that variable is a boolean as I need to update the meta based on either true or false. Using the filter, the if statement won't be accessed if false.

It's quite counter intuitive but:
$ is true? so 'disabled'
$ is false? so 'enabled'